mrmustard/lab_dev/transformations/base.py

```diff
@@ -42,25 +42,69 @@ class Transformation(CircuitComponent):
 
     def inverse(self) -> Transformation:
         r"""Returns the inverse of the transformation."""
+        if len(self.wires.input) != len(self.wires.output):
+            raise NotImplementedError(
+                "Only Transformations with the same number of input and output wires are supported."
+            )
         if not isinstance(self.representation, Bargmann):
             raise NotImplementedError("Only Bargmann representation is supported.")
         if self.representation.ansatz.batch_size > 1:
             raise NotImplementedError("Batched transformations are not supported.")
+
+        # compute the inverse
         A, b, _ = self.dual.representation.conj().triple  # apply X
-        almost_inverse = self.__class__.from_bargmann(
-            [0], (math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 + 0j)
+        almost_inverse = self._from_attributes(
+            Bargmann(math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 + 0j),
+            self.wires,
+            "",
         )
         almost_identity = (
             self >> almost_inverse
         )  # TODO: this is not efficient, need to get c from formula
         invert_this_c = almost_identity.representation.c
-        actual_inverse = self.__class__.from_bargmann(
-            [0], (math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 / invert_this_c)
+        actual_inverse = self._from_attributes(
+            Bargmann(math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 / invert_this_c),
+            self.wires,
+            self.name + "_inv",
         )
         return actual_inverse
 
 
-class Unitary(Transformation):
+class Operator(Transformation):
+    r"""A CircuitComponent with input and output wires, only on the ket side.
+    This class essentially relaxes the requirement that Unitaries have that
+    the input and output modes must be the same."""
```

Comment:
Can you clarify more what you mean by the input and output modes don't need to be the same? E.g. for the BtoQ map, the input and output modes attribute could be the same, it's that the representation is different on the input and output right?

Reply:
No, here I mean that an Operator can have e.g. input on mode [0] and output on modes [0,1]. A Unitary cannot do that.
The other open PR is on top of this one, and I'm working on docstrings in that one because some methods are moving around between the two PRs and I don't want to do things twice.